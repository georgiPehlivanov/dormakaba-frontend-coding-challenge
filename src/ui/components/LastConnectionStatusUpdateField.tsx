import { getLocaleString } from '@/lib/dateTime';
import { Typography } from '@mui/material';

interface LastConnectionStatusUpdateFieldProps {
  timestamp: string;
}

export function LastConnectionStatusUpdateField({
  timestamp,
}: LastConnectionStatusUpdateFieldProps) {
  return (
    <Typography
      display="flex"
      alignItems="center"
      height="100%"
      fontSize="medium"
    >
      {getLocaleString(timestamp)}
    </Typography>
  );
}
