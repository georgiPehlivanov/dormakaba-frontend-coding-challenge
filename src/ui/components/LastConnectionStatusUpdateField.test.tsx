import { render } from '@testing-library/react';
import { LastConnectionStatusUpdateField } from './LastConnectionStatusUpdateField';

const timestamp = '2023-02-22T21:05:12.721Z';

describe('DoorList', () => {
  it('should render correctly', () => {
    const { container } = render(
      <LastConnectionStatusUpdateField timestamp={timestamp} />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
