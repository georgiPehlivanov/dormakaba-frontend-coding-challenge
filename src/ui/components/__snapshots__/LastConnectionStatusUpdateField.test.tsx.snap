// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DoorList should render correctly 1`] = `
<p
  class="MuiTypography-root MuiTypography-body1 css-1dz3uf8-MuiTypography-root"
>
  2/22/2023, 11:05 PM
</p>
`;
